cmake/Modules/CocosBuildHelpers.cmake

```diff
@@ -94,14 +204,104 @@ function(cocos_mark_code_files cocos_target)
   
 endfunction()
 
+# APP_SRC app needed all src files
+# DEPEND_COMMON_LIBS the app needed libs for all platforms
+# DEPEND_ANDROID_LIBS the app needed libs only for android platform
+macro(cocos_mark_app app_name)
```

Comment:
what does `cocos_mark_app ` do? I can not understand it from the macro name. And is it better to use capital characters for macro?

Reply:
input a name, and other application related informations, mark those as a cocos application which name is `app_name`.

this macro reduce much duplicate code, for those content is necessary for every application.  for example mark cpp_empty_test application.
```
# mark app complie info and libs info
cocos_mark_app(${APP_NAME} 
    APP_SRC "${all_code_files}"
    DEPEND_COMMON_LIBS "cocos2d"
    DEPEND_ANDROID_LIBS "cocos2d_android"
)
```

Reply:
meanwhile it's useful to hide the details of using compiled lib or prebuilt lib, it make the application CMakeFiles.txt more clear, like [cpp-template-default/CMakeLists.txt](https://github.com/drelaptop/cocos2d-x/blob/v3-cmake-win32/templates/cpp-template-default/CMakeLists.txt)

macro is quite like a function in cmake, I will try to switch this macro to a function, if I can't I change to use capital chars for macro, and lower-case for function name.